linode_api4/objects/linode.py

```diff
@@ -720,7 +720,13 @@ def reboot(self):
             return False
         return True
 
-    def resize(self, new_type, allow_auto_disk_resize=True, **kwargs):
+    def resize(
+        self,
+        new_type,
+        allow_auto_disk_resize=True,
+        migration_type="cold",
```

Comment:
optional: We can create a class similar to [https://github.com/linode/linode_api4-python/blob/dev/linode_api4/objects/object_storage.py#L14](ObjectStorageACL) to make migration types a bit more explicit.

e.g. 
```python
class MigrationType:
    COLD = "cold"
    WARM = "warm"
```